ios/Classes/FlutterBleLibPlugin.m

```diff
@@ -1,20 +1,155 @@
 #import "FlutterBleLibPlugin.h"
+#import <flutter_ble_lib-Swift.h>
+#import "Constants/ArgumentKey.h"
+#import "Constants/ChannelName.h"
+#import "Constants/MethodName.h"
+#import "Event/AdapterStateStreamHandler.h"
+#import "Event/RestoreStateStreamHandler.h"
+#import "Event/ScanningStreamHandler.h"
+#import "Event/ConnectionStateStreamHandler.h"
+#import "Util/ArgumentValidator.h"
+#import "Util/FlutterErrorFactory.h"
+
+typedef void (^Resolve)(id result);
+typedef void (^Reject)(NSString *code, NSString *message, NSError *error);
+
+@interface FlutterBleLibPlugin () <BleClientManagerDelegate>
+
+@property (nonatomic) BleClientManager *manager;
+@property (nonatomic) AdapterStateStreamHandler *adapterStateStreamHandler;
+@property (nonatomic) RestoreStateStreamHandler *restoreStateStreamHandler;
+@property (nonatomic) ScanningStreamHandler *scanningStreamHandler;
+@property (nonatomic) ConnectionStateStreamHandler *connectionStateStreamHandler;
+
+@end
 
 @implementation FlutterBleLibPlugin
-+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
-  FlutterMethodChannel* channel = [FlutterMethodChannel
-      methodChannelWithName:@"flutter_ble_lib"
-            binaryMessenger:[registrar messenger]];
-  FlutterBleLibPlugin* instance = [[FlutterBleLibPlugin alloc] init];
-  [registrar addMethodCallDelegate:instance channel:channel];
-}
-
-- (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
-  if ([@"getPlatformVersion" isEqualToString:call.method]) {
-    result([@"iOS " stringByAppendingString:[[UIDevice currentDevice] systemVersion]]);
-  } else {
-    result(FlutterMethodNotImplemented);
-  }
+
+// MARK: - Initializers
+
+- (instancetype)init
+{
+    self = [super init];
+    if (self) {
+        self.adapterStateStreamHandler = [AdapterStateStreamHandler new];
+        self.restoreStateStreamHandler = [RestoreStateStreamHandler new];
+        self.scanningStreamHandler = [ScanningStreamHandler new];
+        self.connectionStateStreamHandler = [ConnectionStateStreamHandler new];
+    }
+    return self;
+}
+
+// MARK: - FlutterPlugin implementation
+
++ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar> *)registrar {
+    FlutterMethodChannel *channel = [FlutterMethodChannel methodChannelWithName:CHANNEL_NAME_FLUTTER_BLE_LIB binaryMessenger:[registrar messenger]];
```

Comment:
I left it as `channel` because it was generated this way during flutter plugin creation. I might change it in the future, but would prefer to do it later on, since I know that this might cause conflicts in parallel branches right now.